### PR TITLE
feat(Host): allow hosts to set contribution policy

### PR DESCRIPTION
**Pull request by segfult**
Resolve opencollective/opencollective#3297

### CODE REVIEW

**Comment by segfult** [comment]
Good morning @znarf. i have made the necessary changes. Please take a look and let me know if i need to change anything else.

# Preview: 

first
second

**Review by SudharakaP — COMMENTED** [review]
@segfult : The `package-lock.json` file changes are not relevant to this change I think. 🙂

**Comment by SudharakaP** [comment]
> @segfult : The `package-lock.json` file changes are not relevant to this change I think. slightly_smiling_face

@segfult : What I meant was that the changes to `package-lock.json` should not be checked in as it's not relevant to this change. But the `package-lock.json` file itself should not be deleted. You should just revert the package-lock.json local changes before checking in. 🙂

**Comment by segfult** [comment]
> @segfult : The `package-lock.json` file changes are not relevant to this change I think. slightly_smiling_face

@SudharakaP Thanks for the catch. will make the change :smile:

**Comment by segfult** [comment]
> > @segfult : The `package-lock.json` file changes are not relevant to this change I think. slightly_smiling_face
> 
> @segfult : What I meant was that the changes to `package-lock.json` should not be checked in as it's not relevant to this change. But the `package-lock.json` file itself should not be deleted. You should just revert the package-lock.json local changes before checking in. slightly_smiling_face

@SudharakaP I know, i accidentally deleted it :worried:. i will force push to the previous commit.

**Comment by segfult** [comment]
@SudharakaP removed the package-lock :+1:

File: components/edit-collective/Menu.js [hunk 1]
```diff
@@ -217,6 +218,8 @@ const shouldDisplaySection = (collective, section) => {
  * Displays the menu for the edit collective page
  */
 const EditCollectiveMenu = ({ collective, selectedSection }) => {
+  console.log(collective);
```
Comment (znarf) [inline components/edit-collective/Menu.js:221]: console.log

**Review by znarf — CHANGES_REQUESTED** [review]
It doesn't seem this is matching the "specs" and the "test" defined for the issues.

The Contribution Policy of the Fiscal Host needs to be visible when contributing to a **Collective** hosted by this Fiscal Host.

If the Collective also has a contribution policy, both needs to be visible in the FAQ.

**Comment by segfult** [comment]
Hello @znarf, i have made the changes. I think now it is accurate to the test and spec.

Host policy:
host

Collective contribute flow:
collective

**Comment by segfult** [comment]
Hello @znarf, i have made the necessary changes. I have reused the policies settings form `collective` since Expense Payouts is also being used by 'USER'.

**Comment by znarf** [comment]
@segfult Thank you for the update, having a look.

Could you try to update from master again to fix the "check-langs" test?

**Review by znarf — CHANGES_REQUESTED** [review]
@segfult For Fiscal Hosts, "Policies" should be where "Expenses & Payouts" was, we don't want to move it.

File: components/faqs/ContributeFAQ.js [hunk 2]
```diff
@@ -45,6 +45,22 @@ const ContributeFAQ = ({ collective, ...props }) => (
         </Content>
       </Entry>
     )}
+    {collective.type === 'COLLECTIVE' &&
```
Comment (znarf) [inline components/faqs/ContributeFAQ.js:48]: Testing `collective.type === 'COLLECTIVE'` seems too restrictive. Other valid values could be `FUND`, `EVENT`, `PROJECT` and maybe more in the future. I see no reason to have it.

File: components/faqs/ContributeFAQ.js [hunk 3]
```diff
@@ -45,6 +45,22 @@ const ContributeFAQ = ({ collective, ...props }) => (
...
+      collective.name !== collective.host.name &&
```
Comment (znarf) [inline components/faqs/ContributeFAQ.js:49]: It's cleaner to test by `id` instead of name.

File: pages/contribution-flow.js [hunk 4]
```diff
@@ -321,6 +322,7 @@ const accountFieldsFragment = gqlV2/* GraphQL */ `
     }
     ... on Organization {
       platformFeePercent
+      contributionPolicy
```
Comment (znarf) [inline pages/contribution-flow.js:325]: I don't see the need for this one.

File: components/edit-collective/Menu.js [hunk 5]
```diff
@@ -190,7 +190,7 @@ const sectionsDisplayConditions = {
   [EDIT_COLLECTIVE_SECTIONS.CONNECTED_ACCOUNTS]: c => isHost(c) || isCollective(c),
   [EDIT_COLLECTIVE_SECTIONS.UPDATES]: c => isFeatureAllowed(c, FEATURES.UPDATES),
   [EDIT_COLLECTIVE_SECTIONS.CONVERSATIONS]: c => isFeatureAllowed(c, FEATURES.CONVERSATIONS) && !isFund(c),
-  [EDIT_COLLECTIVE_SECTIONS.POLICIES]: c => isCollective(c) || isFund(c),
+  [EDIT_COLLECTIVE_SECTIONS.POLICIES]: c => isHost(c) || isCollective(c) || isFund(c),
```
Comment (znarf) [inline components/edit-collective/Menu.js:193]: As written elsewhere, this should not change.

File: components/edit-collective/Menu.js [hunk 6]
```diff
@@ -267,7 +267,7 @@ const EditCollectiveMenu = ({ collective, selectedSection }) => {
             renderMenuItem(getSectionInfo(EDIT_COLLECTIVE_SECTIONS.HOST_PLAN))}
           {[USER, ORGANIZATION].includes(collective.type) &&
             renderMenuItem(getSectionInfo(EDIT_COLLECTIVE_SECTIONS.HOST_METRICS))}
-          {[USER, ORGANIZATION].includes(collective.type) &&
+          {[USER].includes(collective.type) &&
```
Comment (znarf) [inline components/edit-collective/Menu.js:270]: Maybe just try to display `POLICIES` here for `ORGANIZATION` and `USER`.

Comment (segfult) [reply components/edit-collective/Menu.js:270]: @znarf thank you for the review, on it. Should i then remove the code related to `EXPENSE_PAYOUTS` cause they aren't used anywhere except here.

File: components/faqs/ContributeFAQ.js [hunk 7]
```diff
@@ -45,6 +45,21 @@ const ContributeFAQ = ({ collective, ...props }) => (
         </Content>
       </Entry>
     )}
+    {(collective.id !== collective.host.id || collective.id === collective.host.id) &&
```
Comment (segfult) [inline components/faqs/ContributeFAQ.js:48]: I added this condition `collective.id === collective.host.id` so the FAQ is displayed when contributing to the `fiscal host` directly too.

File: components/edit-collective/Menu.js (hunk 6 above)
Comment (znarf) [reply components/edit-collective/Menu.js]: If it's not used, sure, let's delete it!

**Review by znarf — APPROVED** [review]

**Comment by znarf** [comment]
Looks good to me!

**Review by Betree — APPROVED** [review]
Looking good!

**Pull request merged**